src/parser/Statement.ts

```diff
@@ -126,7 +152,7 @@ export class ExitWhile extends AstNode implements Statement {
 
 export class Function extends AstNode implements Statement {
     constructor(readonly name: Identifier, readonly func: Expr.Function) {
-        super("Function");
+        super("NamedFunction");
```

Comment:
Needed to change this because `Statement` also has a class called `"Function"`

Reply:
Great catch!